dask_expr/_expr.py

```diff
@@ -2787,7 +2796,20 @@ def _lower(self):
         return None
 
     def _simplify_down(self):
-        if self.freq is None:
+        if self.freq:
```

Comment:
How is this MapPartitions and why don't we need data from previous partitions?

Reply:
Just following the implementation here.. 
https://github.com/dask/dask/blob/0253207f8b3bc2a48e059a856783a4fb7c4eb850/dask/dataframe/core.py#L2197, do you suspect that is wrong?